pkg/pillar/hypervisor/hypervisor.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2017-2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package hypervisor
+
+import (
+	"os"
+)
+
+// Hypervisor provides methods for manipulating domains on the host
+type Hypervisor interface {
+	Name() string
+
+	Create(string, string) (int, error)
+
+	Start(string, int) error
+	Tune(string, int, int) error
+	Stop(string, int, bool) error
+	Delete(string, int) error
+	Info(string, int) error
+	LookupByName(string, int) (int, error)
+
+	IsDeviceModelAlive(int) bool
+
+	PCIReserve(string) error
+	PCIRelease(string) error
+}
+
+// GetHypervisor returns a particular hypervisor implementation
+func GetHypervisor(hint string) Hypervisor {
+	var knownHypervisors = map[string]func() Hypervisor{
+		"xen":  newXen,
+		"null": newNull,
+		"kvm":  newKvm,
+		"acrn": newAcrn,
+	}
+
+	if knownHypervisors[hint] == nil {
```

Comment:
I don't like this part. It is possible to have device files for multiple, and that could create a mess. If you ask `GetHypervisor()` and do not pass it a valid known name, return an error. Do not even have a default; that is at the CLI options level, not at the library level. I would go so far as to have it look like:

```go
func GetHypervisor(hint string) Hypervisor {
	if hyp, ok := knownHypervisors[hint]; ok {
           return hyp(), nil
        }
        return nil, fmt.Errorf("unknown hypervisor: %s", hint)
 }
```

Reply:
So you're saying that we should return an error and let the calling code decide if they want to go with null or something @deitch ?

Reply:
Yes.

Reply:
Done